fem/nonlininteg.hpp

```diff
@@ -68,6 +69,50 @@ class NonlinearFormIntegrator
                                    ElementTransformation &Tr,
                                    const Vector &elfun);
 
+   // TODO: add support for other assembly levels (in addition to PA) and their
+   // actions.
+
+   // TODO: for mixed meshes the quadrature rules to be used by methods like
+   // Setup() can be given as a QuadratureSpace, e.g. using a new method:
+   // SetQuadratureSpace().
+
+   // TODO: the methods for the various assembly levels make sense even in the
+   // base class NonlinearFormIntegrator, except that not all assembly levels
+   // make sense for the action of the nonlinear operator (but they all make
+   // sense for its Jacobian).
+
+   /// Method defining partial assembly.
+   /** The result of the partial assembly is stored internally so that it can be
+       used later in the methods AddMultPA() and AddMultTransposePA(). */
+   virtual void Setup(const FiniteElementSpace &fes);
+
+   /** The result of the partial assembly is stored internally so that it can be
+       used later in the methods AddMultPA() and AddMultTransposePA().
+       Used with BilinearFormIntegrators that have different spaces. */
+   virtual void Setup(const FiniteElementSpace &trial_fes,
+                      const FiniteElementSpace &test_fes);
```

Comment:
I think `Setup` is a good name for these functions. Should `AssemblePA` for the bilinear form integrators be renamed to be consistent with this?

Reply:
Another option is to rename `Setup` to be `AssemblePA`, but I don't prefer that option.

Reply:
No strong feelings here. Since the legacy is AssemblePA, I should probably adhere to that.